Support procedure expire_snapshots for iceberg (#22609)
prestodb/presto: presto-docs/src/main/sphinx/connector/iceberg.rst

    ``unregister_table`` only when using the Hive catalog. This is similar to
    the behavior listed above for the ``DROP TABLE`` command.

Expire snapshots
^^^^^^^^^^^^^^^^

> Each dml action in Iceberg produces a new snapshot while keeping the old data and metadata around for snapshot isolation and time travel. The expire_snapshots procedure can be used to remove older snapshots and their files which are no longer needed.

Reviewer: ```suggestion
Each DML (Data Manipulation Language) action in Iceberg produces a new snapshot while keeping the old data and metadata for snapshot isolation and time travel. Use `expire_snapshots` to remove older snapshots and their files.
```
Suggest capitalizing DML and adding the meaning of the acronym, other suggestions for active voice and conciseness.

Author: Thanks for the suggestion, fixed!

> This procedure will remove old snapshots and the corresponding snapshot files, and it will never remove files which are still required by a non-expired snapshot.

Reviewer: ```suggestion
This procedure removes old snapshots and their corresponding files, and never removes files which are required by a non-expired snapshot.
```

The following arguments are available:

===================== ========== =============== =======================================================================
Argument Name         required   type            Description
===================== ========== =============== =======================================================================
``schema``            ✔️         string          Schema of the table to update

``table_name``        ✔️         string          Name of the table to update

``older_than``                   timestamp       Timestamp before which snapshots will be removed (Default: 5 days ago)

``retain_last``                  int             Number of ancestor snapshots to preserve regardless of older_than
                                                 (defaults to 1)

> ``snapshot_ids``                 array of long   Array of snapshot IDs to expire

Reviewer: Is a word missing after "array of long"?

Author: No, what I want to express is that the type is `array of long`. Is it OK with this expression in your opinion?

Reviewer: That's fine! I was unfamiliar with the phrase and I should have looked it up first. This is good as is. Thanks!

===================== ========== =============== =======================================================================

Examples:

* Remove snapshots older than specific day and time, but retain the last 10 snapshots::

    CALL iceberg.system.expire_snapshots('schema_name', 'table_name', TIMESTAMP '2023-08-31 00:00:00.000', 10);

> * Remove snapshots with snapshot ID 10001 and 10002 (note that this snapshot ID should not be the current snapshot)::

Reviewer: ```suggestion
* Remove snapshots with snapshot ID 10001 and 10002 (note that these snapshot IDs should not be the current snapshot)::
```
